userspace/falco/yaml_helper.h

```diff
@@ -77,22 +78,84 @@ class yaml_visitor {
 class yaml_helper
 {
 public:
+	inline static const std::string configs_key = "configs_files";
+
 	/**
 	* Load the YAML document represented by the input string.
 	*/
 	void load_from_string(const std::string& input)
 	{
 		m_root = YAML::Load(input);
-		pre_process_env_vars();
+		pre_process_env_vars(m_root);
 	}
 
 	/**
 	* Load the YAML document from the given file path.
 	*/
-	void load_from_file(const std::string& path)
+	void load_from_file(const std::string& path, std::vector<std::string>& loaded_config_files)
 	{
-		m_root = YAML::LoadFile(path);
-		pre_process_env_vars();
+		loaded_config_files.clear();
+		m_root = load_from_file_int(path, loaded_config_files);
+
+		const auto ppath = std::filesystem::path(path);
+		const auto config_folder = ppath.parent_path();
+		// Parse files to be included
+		std::vector<std::string> include_files;
+		get_sequence<std::vector<std::string>>(include_files, configs_key);
+		for(const std::string& include_file : include_files)
+		{
+			// If user specifies a relative include file,
+			// make it relative to main config file folder,
+			// instead of cwd.
+			auto include_file_path = std::filesystem::path(include_file);
+			if (!include_file_path.is_absolute())
+			{
+				include_file_path = config_folder / include_file;
+			}
+			if (include_file_path == ppath)
+			{
+				throw std::runtime_error(
+					"Config error: '" + configs_key + "' directive tried to recursively include main config file: " + path + ".");
+			}
+			if (std::filesystem::exists(include_file_path))
+			{
+				if (std::filesystem::is_regular_file(include_file_path))
+				{
+					include_config_file(include_file_path.string(), loaded_config_files);
+				}
+				else if (std::filesystem::is_directory(include_file_path))
+				{
+					std::vector<std::string> v;
+					const auto it_options = std::filesystem::directory_options::follow_directory_symlink
+								| std::filesystem::directory_options::skip_permission_denied;
+					for (auto const& dir_entry : std::filesystem::directory_iterator(include_file_path, it_options))
+					{
+						if (std::filesystem::is_regular_file(dir_entry.path()))
+						{
+							v.push_back(dir_entry.path().string());
+						}
+						// We don't support nested directories
+						else
+						{
+							falco_logger::log(falco_logger::level::WARNING, "Included config file has wrong type: " + dir_entry.path().string());
```

Comment:
NOTE: since we are logging BEFORE the config is actually parsed, we will always log at WARNING level even if user sets log level to `error`.
Otherwise we could return a list of warnings and let the caller manage them. WDYT?

Reply:
>  Otherwise we could return a list of warnings and let the caller manage them.

Done in latest commit. This is the same thing we do with the ruleset so i guessed that's ok!